unit_test/test_main.py

```diff
@@ -148,6 +161,220 @@ def test__cleanup_barrier__cleaning_barrier(self) -> None:
         self.assertNotIn(job_id, barrier_manager._barriers)
 
 
+class OrchestratorTest(unittest.TestCase):
+    def test__new_job_submitted_handler__fully_new_job(self) -> None:
+        # Arrange
+        omotes_orchestrator_if = Mock()
+        jobs_broker_if = Mock()
+        celery_if = Mock()
+        celery_if.start_workflow.return_value = "celery_id"
+        postgresql_if = Mock()
+        postgresql_if.job_exists.return_value = False
+        workflow_manager = Mock()
```

Comment:
duplicate code: lines 173-207 and 233-367, maybe put in function?

Reply:
I found this one difficult. I have reduced it some here: https://github.com/Project-OMOTES/orchestrator/pull/36/commits/f7df22d7b74a85866100012c4f989ba4c704279e

However, most values are required throughout the function so I am afraid that further deduplication will hurt readability of the test. What do you think? Is it reduced enough like this?

Reply:
yes, much better